### ISSUE TITLE
Unable to locate serializer or HTTP client packages under ASP.NET 5

### ISSUE THREAD

**Issue by nbarbettini** [stormpath/stormpath-sdk-dotnet#93]
This bug is similar to #37 but for the ASP.NET 5 (vNext) environment, with IIS Express or `dnx web`.

**Comment by nbarbettini** [comment]
Portions of the plugin architecture were refactored in #102 to mitigate this issue.

If an HTTP client or serializer is not specified when using `IClientBuilder`, some heuristics are used to locate the default HTTP client and serializer packages. These heuristics don't work for the DNX environment, since packages are stored and handled differently.

Instead of building heuristics for every possible environment, the changes in #102 make it easy to manually specify which plugin to use:

```csharp
IClient client = Clients.Builder()
    .SetHttpClient(HttpClients.Create().RestSharpClient())
    .SetSerializer(Serializers.Create().JsonNetSerializer())
    // ...
    .Build();
```

**tl;dr** In most cases, it isn't necessary to specify the HTTP client and serializer plugin when building a `IClient`. But if the SDK complains about not being able to find a plugin, simply use the methods available in `HttpClients.Create()...` and `Serializers.Create()...` to manually specify which package to use.

**Comment by DarinMacRae** [comment]
I just got hit by this and am not sure why. It would have been helpful and saved me time if this was documented as part of the quick start guide.

(It could be explained by the code appearing to only "auto-configure" if running in .NET 4.5. I'm running .NET 4.6. If this is the reason why not support the latest?)

https://github.com/stormpath/stormpath-sdk-dotnet/blob/28f40341249d127b31532d2d6962140ecfd7e5d2/src/Stormpath.SDK.Core/Impl/Utility/TypeLoader.cs#L38-L46

**Comment by nbarbettini** [comment]
Sorry about the frustration @DarinMacRae! Trying to auto-detect the platform and pull in the right plugins ended up being more trouble than it was worth, and I'm currently planning on removing it entirely for version 1.0. It'll be a breaking change, though, hence why it's waiting for a major version increment.

FWIW, if you are using our ASP.NET integration ([repo](https://github.com/stormpath/stormpath-aspnet)), everything should be configured for you automatically (and correctly).
